File: atcoder/abc197d.py
import math
import cmath

N = int(input())
x0, y0 = map(float, input().split())
xN2, yN2 = map(float, input().split())

p0 = complex(x0, y0)
pN2 = complex(xN2, yN2)
o = (p0+pN2)/2
r = cmath.rect(1, 2*math.pi/N)
a = (p0-o)*r+o

print("%.10f %.10f" % (a.real, a.imag))

File: atcoder/abc197d.cpp

#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    double x, y;
    complex<double> a, b;

    cin >> n;
    cin >> x >> y;
    a = {x, y};
    cin >> x >> y;
    b = {x, y};

    auto m = (a + b) / 2.0;

    auto res = m + (a - m) * polar(1.0, M_PI * 2.0 / n);

    cout << setprecision(9) << res.real() << " " << res.imag() << endl;
}
